controllers/grid/files/proof/ManageProofFilesGridHandler.inc.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * @file controllers/grid/files/proof/ManageProofFilesGridHandler.inc.php
+ *
+ * Copyright (c) 2014-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ManageProofFilesGridHandler
+ * @ingroup controllers_grid_files_proof
+ *
+ * @brief Handle the editor's proof files selection grid (selects which files to include)
+ */
+
+import('lib.pkp.controllers.grid.files.SelectableSubmissionFileListCategoryGridHandler');
+
+class ManageProofFilesGridHandler extends SelectableSubmissionFileListCategoryGridHandler {
+	/**
+	 * Constructor
+	 */
+	function ManageProofFilesGridHandler() {
+		import('controllers.grid.files.proof.ProofFilesCategoryGridDataProvider');
+		parent::SelectableSubmissionFileListCategoryGridHandler(
+			new ProofFilesCategoryGridDataProvider(),
+			WORKFLOW_STAGE_ID_PRODUCTION,
+			0
+		);
+
+		$this->addRoleAssignment(
+			array(
+				ROLE_ID_SUB_EDITOR,
+				ROLE_ID_MANAGER,
+				ROLE_ID_ASSISTANT
+			),
+			array(
+				'fetchGrid', 'fetchCategory', 'fetchRow',
+				'addFile',
+				'downloadFile',
+				'deleteFile',
+				'updateProofFiles'
+			)
+		);
+
+		// Set the grid title.
+		$this->setTitle('submission.pageProofs');
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.internal.SubmissionRequiredPolicy');
+		$this->addPolicy(new SubmissionRequiredPolicy($request, $args, 'submissionId'));
+
+		import('classes.security.authorization.internal.PublicationFormatRequiredPolicy');
+		$this->addPolicy(new PublicationFormatRequiredPolicy($request, $args));
+		return parent::authorize($request, $args, $roleAssignments);
```

Comment:
The right authorization to use here is SubmissionAccessPolicy instead of only the required one. The access policy will really check if the user has access to the submission, the required one will only make sure we have a valid submission in request.